spmd/tensor/parallel/api.py

```diff
@@ -76,12 +76,69 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 module.register_module(n, tp_multi_head_attention)
 
 
+def _isMLP(module: nn.Module) -> bool:
+    # We assume that the structure of MLP to shard defined in object way as below:
+    # Linear -> ReLU -> Linear
+    submodules = list(module.children())
+    return (
+        len(submodules) == 3
+        and isinstance(submodules[0], nn.Linear)
+        and isinstance(submodules[1], nn.ReLU)
+        and isinstance(submodules[2], nn.Linear)
+    )
+
+
+def _gradient_hook(param: DTensor, grad: DTensor) -> None:
+    param._local_tensor.grad = grad._local_tensor
+
+
+def _shard_mlp(name: str, module: nn.Module, device_mesh: DeviceMesh) -> None:
+    if _isMLP(module):
```

Comment:
If it's not a MLP module, maybe we raise a warning here? so that user know that they are not using this API properly.

Reply:
@wanchaol We may need to raise a warning from a different place because this `_shard_mlp` function will be applied to all nodes in the module tree. One question about the usage of `distribute_module()`: 

1. Does user write `distribute_module(model, parallel_fn=tp_shard_mlp)` to parallelize all MLP submodules in model, or 
2. User write `distribute_module(MLP, parallel_fn=tp_shard_mlp)` to parallelize one MLP layer.

I think it is the first case, then we should not raise warning from `_shard_mlp` but some function at top-level node of the tree.